packages/vizbuilder/src/index.js

```diff
@@ -0,0 +1,198 @@
+  }
+
+  getChildContext() {
+    return {
+      loadControl: this.loadControl,
+      stateUpdate: this.stateUpdate
+    };
+  }
+
+  componentDidMount() {
+    const locationQuery = queryString.parse(location.search);
+    this.firstLoad(locationQuery);
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    const {src} = this.props;
+    const {load} = this.state;
+    const {error, severity} = load;
+
+    if (src && prevProps.src !== src) {
+      api.resetClient(src);
+      this.setState(initialState(), this.firstLoad);
+    }
+
+    if (error && prevState.load.error !== error) {
+      console.warn(error.stack);
+      UIToaster.show({intent: severity, message: error.message});
+    }
+  }
+
+  render() {
+    const {config} = this.props;
+    const {dataset, load, members, options, query, queryOptions} = this.state;
+    return (
+      <div className={classnames("vizbuilder", {loading: load.inProgress})}>
+        <LoadingScreen {...load} />
+        <Sidebar
+          options={options}
+          query={query}
+          queryOptions={queryOptions}
+        />
+        <ChartArea
+          dataset={dataset}
+          members={members}
+          query={query}
+          userConfig={config}
+        />
+      </div>
+    );
+  }
+
+  stateUpdate(newState) {
+    return setStatePromise.call(this, state => {
+      const finalState = {};
+      const keys = Object.keys(newState);
+
+      for (let i = 0; i < keys.length; i++) {
+        const key = keys[i];
+        finalState[key] = {
+          ...state[key],
+          ...newState[key]
+        };
+      }
+
+      return finalState;
+    });
+  }
+
+  firstLoad(locationQuery) {
+    this.loadControl(fetchCubes.bind(this, locationQuery), () => {
+      const {query, queryOptions} = this.state;
+      return api.query({
+        ...query,
+        options: queryOptions
+      });
+    });
+  }
+}
+
+Vizbuilder.childContextTypes = {
+  loadControl: PropTypes.func,
+  stateUpdate: PropTypes.func
+};
+
+Vizbuilder.propTypes = {
+  config: PropTypes.shape({
+    topojson: PropTypes.objectOf(
+      PropTypes.shape({
+        topojson: PropTypes.string.isRequired,
+        topojsonId: PropTypes.string,
+        topojsonKey: PropTypes.string
+      })
+    ),
+    chartConfig: PropTypes.shape({
+      common: PropTypes.object,
+      barchart: PropTypes.object,
+      barchartActive: PropTypes.object,
+      barchartyear: PropTypes.object,
+      barchartyearActive: PropTypes.object,
+      donut: PropTypes.object,
+      donutActive: PropTypes.object,
+      geomap: PropTypes.object,
+      geomapActive: PropTypes.object,
+      lineplot: PropTypes.object,
+      lineplotActive: PropTypes.object,
+      stacked: PropTypes.object,
+      stackedActive: PropTypes.object,
+      treemap: PropTypes.object,
+      treemapActive: PropTypes.object
+    })
+  }),
+  src: PropTypes.string.isRequired
+};
+
+// EXAMPLE, will be removed later
+Vizbuilder.defaultProps = {
```

Comment:
I tried a simple copy/paste of the vizbuilder into my Data USA test site, and am having problems overriding some of these configs. I suggest we make a couple of changes:
1. a property called `topojson`, which like you have it, accepts keys for cube `Level` names. The configuration for each level should simply be folded into the default config that is constructed for Geomaps: `{..geoMapConfig, ...topojsonConfig[level]}`. Default should be `{}`, which means no Geomaps out of the box because they rely on static topojson and we can't predict where the user could be storing it.
2. a property called `config`, which is simply a standard d3plus config object (but defaults to and empty `{}` object). Most configs will/should be set in `app/d3plus.js`, which gets applied to all `d3plus-react` components automatically through context, but I can see a scenario where the vizbuilder may need unique styles
3. a property called `visualizations`, which should be an array of string values matching the visualization names defined in your `charts` variable. this would allow the user to toggle the visibility of specific visualization types. By default, do not show `Pie` or `Donut` (`Treemap` is our preferred "share" visualization)